Unit test failures in drawing_classifier

Repro steps:
```
pytest test/test_drawing_classifier.py
```

```
=================================== FAILURES ===================================
____________ DrawingClassifierTest.test_evaluate_with_ground_truth _____________

self = <turicreate.test.test_drawing_classifier.DrawingClassifierTest testMethod=test_evaluate_with_ground_truth>

    def test_evaluate_with_ground_truth(self):
    
        all_metrics = ["accuracy", "auc", "precision", "recall",
                       "f1_score", "log_loss", "confusion_matrix", "roc_curve"]
        for index in range(len(self.models)):
            model = self.models[index]
            sf = self.trains[index]
            individual_run_results = dict()
            for metric in all_metrics:
>               evaluation = model.evaluate(sf, metric=metric)

test_drawing_classifier.py:241: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../toolkits/drawing_classifier/drawing_classifier.py:725: in evaluate
    'recall', 'auc', 'roc_curve', 'confusion_matrix']}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <listiterator object at 0x7faeb4655550>

>   evaluation_result = {k: ret[k] for k in ['accuracy', 'f1_score', 'precision',
                                             'recall', 'auc', 'roc_curve', 'confusion_matrix']}
E   KeyError: 'f1_score'

../toolkits/drawing_classifier/drawing_classifier.py:724: KeyError
----------------------------- Captured stdout call -----------------------------
Predicting 11/10
_______ DrawingClassifierFromScratchTest.test_evaluate_with_ground_truth _______

self = <turicreate.test.test_drawing_classifier.DrawingClassifierFromScratchTest testMethod=test_evaluate_with_ground_truth>

    def test_evaluate_with_ground_truth(self):
    
        all_metrics = ["accuracy", "auc", "precision", "recall",
                       "f1_score", "log_loss", "confusion_matrix", "roc_curve"]
        for index in range(len(self.models)):
            model = self.models[index]
            sf = self.trains[index]
            individual_run_results = dict()
            for metric in all_metrics:
>               evaluation = model.evaluate(sf, metric=metric)

test_drawing_classifier.py:241: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../toolkits/drawing_classifier/drawing_classifier.py:725: in evaluate
    'recall', 'auc', 'roc_curve', 'confusion_matrix']}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <listiterator object at 0x7faeb45678d0>

>   evaluation_result = {k: ret[k] for k in ['accuracy', 'f1_score', 'precision',
                                             'recall', 'auc', 'roc_curve', 'confusion_matrix']}
E   KeyError: 'f1_score'

../toolkits/drawing_classifier/drawing_classifier.py:724: KeyError
----------------------------- Captured stdout call -----------------------------
Predicting 11/10
______ DrawingClassifierUsingQuickdraw245.test_evaluate_with_ground_truth ______

self = <turicreate.test.test_drawing_classifier.DrawingClassifierUsingQuickdraw245 testMethod=test_evaluate_with_ground_truth>

    def test_evaluate_with_ground_truth(self):
    
        all_metrics = ["accuracy", "auc", "precision", "recall",
                       "f1_score", "log_loss", "confusion_matrix", "roc_curve"]
        for index in range(len(self.models)):
            model = self.models[index]
            sf = self.trains[index]
            individual_run_results = dict()
            for metric in all_metrics:
>               evaluation = model.evaluate(sf, metric=metric)

test_drawing_classifier.py:241: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../toolkits/drawing_classifier/drawing_classifier.py:725: in evaluate
    'recall', 'auc', 'roc_curve', 'confusion_matrix']}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <listiterator object at 0x7faeb44e8c50>

>   evaluation_result = {k: ret[k] for k in ['accuracy', 'f1_score', 'precision',
                                             'recall', 'auc', 'roc_curve', 'confusion_matrix']}
E   KeyError: 'f1_score'

../toolkits/drawing_classifier/drawing_classifier.py:724: KeyError
----------------------------- Captured stdout call -----------------------------
Predicting 11/10
```

--- a/src/python/turicreate/test/test_drawing_classifier.py
+++ b/src/python/turicreate/test/test_drawing_classifier.py
@@ -220,7 +220,6 @@ class DrawingClassifierTest(unittest.TestCase):
                 preds = model.predict(sf[self.feature], output_type="probability")
                 assert (preds.dtype == float)
 
-
     def test_evaluate_without_ground_truth(self):
         for index in range(len(self.trains)):
             model = self.models[index]
@@ -229,9 +228,7 @@ class DrawingClassifierTest(unittest.TestCase):
             with self.assertRaises(_ToolkitError):
                 model.evaluate(sf_without_ground_truth)
 
-    @pytest.mark.xfail(reason='Fails, see https://github.com/apple/turicreate/issues/1973')
     def test_evaluate_with_ground_truth(self):
-
         all_metrics = ["accuracy", "auc", "precision", "recall",
                        "f1_score", "log_loss", "confusion_matrix", "roc_curve"]
         for index in range(len(self.models)):

--- a/src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py
+++ b/src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py
@@ -728,9 +728,7 @@ class DrawingClassifier(_CustomModel):
             hclusterSort,
             l2Dist
         )
-
-        evaluation_result = {k: ret[k] for k in ['accuracy', 'f1_score', 'precision',
-                                                 'recall', 'auc', 'roc_curve', 'confusion_matrix']}
+        evaluation_result = {k: ret[k] for k in metrics}
         evaluation_result['num_test_examples'] = len(dataset)
         for k in ['num_classes', 'num_examples', 'training_loss', 'training_time', 'max_iterations']:
             evaluation_result[k] = getattr(self, k)

